server/tserver/src/main/java/org/apache/accumulo/tserver/replication/AccumuloReplicaSystem.java

```diff
@@ -288,15 +295,36 @@ public String execute(ReplicationCoordinator.Client client) throws Exception {
             }
           } else {
             span = Trace.start("WAL replication");
+
+            ExecutorService executor = Executors.newFixedThreadPool(1);
+
             try {
-              finalStatus = replicateLogs(peerContext, peerTserver, target, p, status, sizeLimit, remoteTableId, peerContext.rpcCreds(), helper, accumuloUgi);
+              Future<Status> replStatus = executor.submit(new Callable<Status>() {
+                @Override
+                public Status call() throws Exception {
+                  return replicateLogs(peerContext, peerTserver, target, p, status, sizeLimit, remoteTableId, peerContext.rpcCreds(), helper, accumuloUgi);
+                }
+              });
+
+              log.debug("Getting replication status with timeout {}", conf.get(Property.REPLICATION_TIMEOUT));
+              finalStatus = replStatus.get(conf.getTimeInMillis(Property.REPLICATION_TIMEOUT), MILLISECONDS);
+              log.debug("New status for {} after replicating to {} is {}", p, peerContext.getInstance(), ProtobufUtil.toString(finalStatus));
+            } catch (InterruptedException e) {
+              log.debug("Interrupted exception during replication", e);
+              Thread.currentThread().interrupt();
+              finalStatus = status;
+            } catch (ExecutionException e) {
+              log.warn("Caught execution exception", e);
+              finalStatus = status;
+            } catch (TimeoutException e) {
+              log.debug("Replication timeout triggered, task will be retried by the framework");
+              finalStatus = status;
             } finally {
               span.stop();
+              executor.shutdownNow();
```

Comment:
Could add comment here that shutdownNow was called instead of shutdown because it will interrupt threads.

If you had an atomic boolean for interrupting, could set it here.